### ISSUE TITLE
Add a CI job with `torch.default_dtype = torch.float64`

### ISSUE THREAD

**Issue by ev-br** [scipy/scipy#22210]
To summarize the discussion from 
https://discuss.scientific-python.org/t/should-scipy-tests-assume-float64-as-the-default-floating-point-dtype/1606/8 :

- in our testing with multiple array backends, we currently configure jax to default to float64 but keep torch defaulting to float32;
- the ability to run scipy tests with either f32 or f64 default dtype is important for some of us;
- we broadly agree that inconsistent defaults are not by design;
- there is a body of tests with handcrafted tolerances depending on whether the array backend is torch or something else.

The path forward---summarizing from and slightly extending the discourse thread above:

- [x] parametrize the array API CI job into a pair: one job sets `torch.set_default_dtype(torch.float64)`, the other does not
- [x] add a helper (`_lib` feels like a right place?) to query the default, roughly `_lib.default_fp_type(xp)` which returns the `xp`-specific default dtype; do not add a publicly accessible backend-agnostic way to set the default.
- [ ] audit existing tests for backend-specific tolerances, `atol=1e-7 if is_torch(xp) else 1e-15`, change to `if default_fp_type(xp) == xp.float32` where appropriate;
- [x] once done, make float64 the default in conftest.py

**Comment by rgommers** [comment]
> in our testing with multiple array backends, we configure jax to default to float64 but keep torch defaulting to float32;

This seems like the wrong take-away (unless you meant "for now, until we have the next items on this list done"), and inconsistent with the rest of the action items? The default should be `float64` for all libraries was my conclusion from the thread.

**Comment by ev-br** [comment]
Yes, thanks. The word "currently" was missing from "we currently configure jax to default to float64 but keep torch defaulting to float32". Edited it in.

**Comment by ev-br** [comment]
The CI job was added in gh-22214, the majority of uses has been audited, and there remaining ones, if any, can be taken care of during regular maintenance.
